core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -620,12 +617,67 @@ class CivilizationInfo {
 
             if (flagsCountdown[flag]!! > 0)
                 flagsCountdown[flag] = flagsCountdown[flag]!! - 1
-
+            
+            if (flagsCountdown[flag]!! != 0) continue
+            
+            when (flag) {
+                CivFlags.TurnsTillNextDiplomaticVote.name -> addFlag(CivFlags.ShowDiplomaticVotingResults.name, 1)
+                CivFlags.ShouldResetDiplomaticVotes.name -> {
+                    gameInfo.diplomaticVictoryVotesCast.clear()
+                    removeFlag(CivFlags.ShouldResetDiplomaticVotes.name)
+                    removeFlag(CivFlags.ShowDiplomaticVotingResults.name)
+                }
+                CivFlags.ShowDiplomaticVotingResults.name -> {
+                    addFlag(CivFlags.ShouldResetDiplomaticVotes.name, 1)
+                    
+                    if (gameInfo.civilizations.any { it.victoryManager.hasWon() } )
+                        // We have either already done this calculation, or it doesn't matter anymore, 
+                        // so don't waste resources doing it
+                        continue
+                    
+                    val results = victoryManager.calculateDiplomaticVotingResults(gameInfo.diplomaticVictoryVotesCast)
+                    val bestCiv = results.maxByOrNull { it.value }
+                    if (bestCiv == null) continue
+                    
+                    val civCount = gameInfo.civilizations.count { !it.isDefeated() }
+                    
+                    // CvGame.cpp::DoUpdateDiploVictory() in the source code of the original
+                    val votesNeededToWin = (
+                        if (civCount > 28) 0.35 * civCount
+                        else (67 - 1.1 * civCount) / 100 * civCount
+                        ).toInt()
```

Comment:
Surely this should be a function, since we'll want to display the vote needed to the player somewhere